### PR TITLE
Add support for the name option

### LINKED ISSUE #227: Support dynamic names
I have the usecase that I need to start ChromicPDF under a supervision tree that is started multiple times (at runtime) under different names.

If I'm not mistaken, currently it's not possible to start ChromicPDF multiple times using a different name.

Are there plans to support this? (I could work on this) To keep things simple for the users, we could adopt the same approach as Ecto (with put_dynamic_repo/1).

### PR DESCRIPTION
**Pull request by dvic**
Closes #227

### CODE REVIEW

**Comment by maltoe** [comment]
@dvic don't worry about the `capture_log` CI failure, it's been flaky ever since we moved to Github actions and we didn't take the time to investigate yet :shrug:

**Comment by dvic** [comment]
> 

Alright, I just pushed my last change, forgot to remove the `is_pid` guard. It should be good to go now.

**Review by maltoe — COMMENTED** [review]
Looking great! :100: Thank you for the contribution!

Also said in the one comment, I wonder if we should move the whole thing to `ChromicPDF.Supervisor` to reduce the noise in the "main" docs. IMO it kind of fits better there, wdyt?

File: lib/chromic_pdf.ex [hunk 1]
```diff
@@ -25,6 +25,29 @@ defmodule ChromicPDF do
         end
       end
 
+  ### Running under a supervision tree
```
[the remark this answers is not in the document]
Comment (dvic) [reply lib/chromic_pdf.ex:28]: And what about moving the whole example to the documentation of `put_dynamic_name/1`? I think that's the best of both worlds.

File: lib/chromic_pdf/supervisor.ex [hunk 2]
```diff
@@ -676,6 +699,23 @@ defmodule ChromicPDF.Supervisor do
       def print_to_pdfa(source, opts \\ []) do
         with_services(__MODULE__, &API.print_to_pdfa(&1, source, opts))
       end
+
+      @doc """
+      Retrieves the currently set name (set using `put_dynamic_name/1`) or the default name.
+      """
+      def get_dynamic_name do
```
[the remark this answers is not in the document]
Comment (dvic) [reply lib/chromic_pdf/supervisor.ex:706]: I'll add a spec, it's just atom() now.

**Comment by maltoe** [comment]
> Also said in the one comment, I wonder if we should move the whole thing to `ChromicPDF.Supervisor` to reduce the noise in the "main" docs. IMO it kind of fits better there, wdyt?

To be precise:

* move both `get_dynamic_name/1` and `put_dynamic_name1` outside of the `__using__` block in `Supervisor`
* move the documentation to the moduledoc in `Supervisor`

**Comment by maltoe** [comment]
> Also said in the one comment, I wonder if we should move the whole thing to `ChromicPDF.Supervisor` to reduce the noise in the "main" docs. IMO it kind of fits better there, wdyt?

To be precise:

* move both `get_dynamic_name/1` and `put_dynamic_name1` outside of the `__using__` block in `Supervisor`
* move the documentation to the moduledoc in `Supervisor`

File: lib/chromic_pdf/supervisor.ex [hunk 3]
```diff
@@ -676,6 +699,23 @@ defmodule ChromicPDF.Supervisor do
...
+        Process.get({__MODULE__, :dynamic_name}, __MODULE__)
+      end
+
+      @doc """
+      Activate a particular ChromicPDF instance, which was started with the `name` option.
+      After calling this function, all calls in the current process will use this instance of ChromicPDF.
+
+      Returns the previously set name or the default name.
+      """
+      def put_dynamic_name(name) when is_atom(name) do
```
[the remark this answers is not in the document]
Comment (dvic) [reply lib/chromic_pdf/supervisor.ex:716]: I think you need this when you have code that needs to temporarily replace the dynamic name and restore it once done. This is at least what Ecto uses it for.

[the remark this answers is not in the document]
Comment (dvic) [reply lib/chromic_pdf/supervisor.ex:716]: Although you could technically just call `get_dynamic_name` in the first place, but it's less convenient.

File: lib/chromic_pdf.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (maltoe) [reply lib/chromic_pdf.ex:28]: Go for it :+1:

File: lib/chromic_pdf/supervisor.ex (hunk 3 above)
[the remark this answers is not in the document]
Comment (maltoe) [reply lib/chromic_pdf/supervisor.ex:716]: if Ecto does it, it must be good :+1: :upside_down_face:

**Comment by dvic** [comment]
> 

I moved the docs to `put_dynamic_name/1`. However, I kept it inside the `__using__` macro because I think it makes more sense to keep it there as the default name is the module that called `use`. Wdyt?

**Review by maltoe — APPROVED** [review]
Awesome! Thank you.

I'll cut a new release later.

**Pull request merged**